ethereum/core/src/test/java/org/hyperledger/besu/ethereum/MainnetBlockValidatorTest.java

```diff
@@ -223,4 +223,48 @@ public void shouldReturnBadBlockBasedOnTheHash() {
         HeaderValidationMode.DETACHED_ONLY);
     assertThat(badBlockManager.getBadBlock(badBlock.getHash())).containsSame(badBlock);
   }
+
+  @Test
+  public void shouldNotAddBadBlockToBadBlockManagerWhenShouldNotRecordBadBlockFlagIsSetToTrue() {
```

Comment:
that's quite a mouthful!

Reply:
```suggestion
  public void when_shouldRecordBadBlockIsFalse_expect_shouldNotAddToBadBlockManager() {
```

Reply:
was looking at this https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea